shpc/main/wrappers/__init__.py

```diff
@@ -22,40 +22,54 @@ def generate(image, container, config, **kwargs):
     settings = container.settings
     aliases = kwargs.get("aliases", [])
 
-    # Global wrapper scripts are generated via aliases
-    for container_tech in ["docker", "podman", "singularity"]:
-        template_name = settings.wrapper_scripts.get(container_tech)
-        if template_name and aliases and container_tech == container.command:
-            wrapper = get_wrapper_script(template_name)(
-                settings=settings,
-                wrapper_template=template_name,
-                config=config,
-                container=container,
-                image=image,
-                **kwargs
-            )
-            generated += wrapper.generate_aliases()
+    constructor_kwargs = {
```

Comment:
I think it would still be nice to have a get_wrapper_script function - e.g., it's different from generate (which is going to write it) but just goes through the logic to return the thing we intend to parse.  This also will make it easier to write tests for different scenarios of getting the script (which I think we likely need!)

Reply:
I've given this a shot in the latest commits (cf `find_wrapper_script` in `base.py`). The function should be easy to test as it doesn't require any of the Jinja stuff of `load_template`. Let me know if that's what you had in mind. Then I can write the tests / doc

Reply:
> 1. Why do we set `self.wrapper_template` and then provide it to the function? It should probably be one or the other - either set it (and use it in the function without providing as an argument) or provide to the function and allow it to be set there.

Ah yes, I've gone back and forth about this and I can see there are remnants of both approaches. Initially I wanted to  make it an argument of the function because it's not much used in other places, but then I thought it'd be good for the class to keep track of the most important stuff, hence the `self.`.

_But_ I've also been thinking about the purpose of this module `shpc/main/wrappers/base.py`. It has a single class, `WrapperScript`, and by making `wrapper_template` a function argument, the class essentially becomes a holder of settings, configuration, overall parameters, etc, – nothing alias-specific –, that has methods to find and load templates, and generate the scripts. And all the alias-specific stuff would be handled as _parameters_ of these methods only. Therefore, there wouldn't be a need to instantiate a new `WrapperScript` for each alias / wrapper script in `shpc/main/wrappers/__init__.py`. I may then rename the class to say `WrapperScriptGenerator`, and the file to say `shpc/main/wrappers/generator.py`, and instantiate it just once in `shpc/main/wrappers/__init__.py:generate`.

What would you think of that ?

Otherwise, I may come back to the opposite approach, and reinstate `wrapper_template` as a mandatory constructor argument 

> 2. Does `self.template_file` need to be used anywhere, other than providing the directory path at the end of the function?

It's its sole use, and I've already made it a local variable instead in a commit I haven't pushed yet.

> 3. if we are returning the template directories (and not the wrapper script, which is set) should it be called `set_wrapper_script`?

I'll probably change the name indeed, also depending of 1) and 4)

> 4. I usually like one function to do one thing - is there a way to piece apart finding the files vs. the directories?

Absolutely

Reply:
Mostly done (I still need to test that it works 😅)